api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -124,8 +125,25 @@ message DownstreamTlsContext {
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
 }
 
+// TLS key log configuration.
+// The key log file format is "format used by NSS for its SSLKEYLOGFILE debugging output" (text taken from openssl man page)
+message TlsKeyLog {
+  // The path to save the TLS key log.
+  string tls_keylog_path = 1 [(validate.rules).string = {min_len: 1}];
+
+  // At least one of src or dst must be specified, or no key logs will be saved for any connections.
```

Comment:
Instead of "or no key logs will be saved for any connections", please say "or the config will be rejected as invalid".  And make sure that the implementation is actually doing that.

I think it's inherently non-sensical to specify a config for key logging that does not actually log any keys.  If someone does that, we should reject it as an error, since that will be much less surprising behavior.

Reply:
@markdroth ,  I can understand your concern. But it seems that currently the protoc-gen-validate tools can not support such validations.

Reply:
You are correct that PGV cannot do cross-fields validations.
However, you can throw an `EnvoyException` in case the condition does not hold, for example:
https://github.com/envoyproxy/envoy/blob/9cc74781d818aaa58b9cca9602fe8dc62181d27b/source/common/ssl/tls_certificate_config_impl.cc#L49-L52

Reply:
Thanks @adisuissa. I will fix it.